src/handler/sign_tx.c

```diff
@@ -65,29 +65,46 @@ bool verify_change(change_output_info_t *info, tx_output_t output) {
  *
  * The outputs will be on the global context for sign tx (`tx_info`)
  **/
-void read_change_info_v1(buffer_t *cdata) {
+uint16_t read_change_info_v1(buffer_t *cdata) {
```

Comment:
We are changing these methods to return an error code instead of throwing one because this method is called from `receive_data` which is outside the `decode_elements` try..catch.

This means that the error is raised to the try..catch of the main function and just returns the error code to the user without cleaning the global context.

The reason this needs to change is because the next call will reach the device while the global context is initialized, and depending on the call it may return a bad state error (even on a perfectly correct call).